docs/user_manual.md

```diff
@@ -49,40 +58,256 @@ python -m virtualenv venv
 Install:
 ``` bash
 # From PyPI:
-pip install datumaro
+pip install datumaro[default]
 
 # From the GitHub repository:
-pip install 'git+https://github.com/openvinotoolkit/datumaro'
+pip install 'git+https://github.com/openvinotoolkit/datumaro[default]'
 ```
 
-> You can change the installation branch with `...@<branch_name>`
-> Also use `--force-reinstall` parameter in this case.
+Read more about choosing between `datumaro` and `datumaro[default]`
+[here](#core-install).
 
-## Interfaces
+**Plugins**
 
-As a standalone tool:
+Datumaro has many plugins, which are responsible for dataset formats,
+model launchers and other optional components. If a plugin has dependencies,
+they can require additional installation. You can find the list of all the
+plugin dependencies in the [plugins](#extending) section.
 
-``` bash
-datum --help
-```
+**Customizing installation parameters**
+
+- <a id="core-install"></a>In restricted installation environments,
+  or if you need only basic Datumaro functionality, you can choose
+  the installation option:
+  - `pip install datumaro` - for core library functionality
+  - `pip install datumaro[default]` - for normal CLI experience
+
+- In some cases, there can be limited use for UI elements outside CLI,
+  or limited options of installing graphical libraries in the system
+  (various Docker environments, servers etc). You can select between using
+  `opencv` and `opencv-headless` by setting the `DATUMARO_HEADLESS`
+  environment variable to `0` or `1` before installing the package.
+  It requires installation from the sources:
+  ```bash
+  DATUMARO_HEADLESS=1 pip install datumaro --no-binary=datumaro
+  ```
+  This option can't be covered by extras due to `pip` limitations.
+
+- Although Datumaro has `pycocotools==2.0.1` in requirements, it works with
+  2.0.2 perfectly fine. The reason for such requirement is binary
+  incompatibility of the `numpy` dependency in the `TensorFlow` and
+  `pycocotools` binary packages
+  (see [#253](https://github.com/openvinotoolkit/datumaro/issues/253))
 
-As a python module:
-> The directory containing Datumaro should be in the `PYTHONPATH`
-> environment variable or `cvat/datumaro/` should be the current directory.
+- You can change the installation branch with `...@<branch_name>`.
```

Comment:
This should probably clarify that it applies in the case of installing directly from the repository.

(Although I'm not sure that this item is needed at all, since it's just a feature of pip and not something specific to Datumaro.)